libqtile/widget/image.py

```diff
@@ -94,3 +97,8 @@ def calculate_length(self):
             return self.img.width + (self.margin_x * 2)
         else:
             return self.img.height + (self.margin_y * 2)
+
+    def cmd_update(self, filename):
+        self.filename = filename
+        self._update_image()
+        self.draw()
```

Comment:
If your new image is not the same size as the previous one then you should do a `self.bar.draw()` to ensure the other widgets' positions are also adjusted.

Reply:
added update for length check and redraw but got some check issue what I don't understand

Reply:
Those check issues aren't caused by this. Still trying to track that one down...

Reply:
If you rebase this branch onto the latest master then those failing tests should disappear.

Reply:
rebased